src/discord/commands/book.ts

```diff
@@ -0,0 +1,227 @@
+
+    for (let i = 0; i < 14; i++) {
+        dateOptions.push({
+            label: `${currentDate.weekdayLong} - ${currentDate.monthLong} ${currentDate.day}${dateSuffix(currentDate.day)}`,
+            value: currentDate.toISODate(),
+            default: currentDate.toISODate() === selectedDate ? true : false,
+        });
+        currentDate = currentDate.plus({ days: 1 });
+    }
+    return dateOptions;
+}
+
+async function parseCommandOptions(interaction: CommandInteraction): Promise<string[] | undefined> {
+    //Parses the book command option parameters to return the corresponding Room's section ID
+    const roomName = interaction.options.getString('room-name');
+    const roomJson = await roomModel.findOne({ name: roomName !== null ? roomName : undefined });
+
+    if (roomJson === null) {
+        interaction.reply({ content: 'Invalid Room: Do `/rooms` to see all avaliable rooms!', ephemeral: true });
+        return undefined;
+    }
+
+    const sectionJson = await sectionModel.find({ roomId: roomJson!._id });
+    const specificSection = sectionJson.find((s) => s.name === interaction.options.getString('section-name'));
+
+    if (specificSection === undefined) {
+        interaction.reply({ content: 'Invalid Section: Do `/rooms` to see all avaliable rooms and sections!', ephemeral: true });
+        return undefined;
+    }
+
+    return [roomJson._id, specificSection._id];
+}
+
+async function searchTimeblocks(selectedDate: string, sectionInformation: SectionInformation, roomInformation: Room) {
+    //Function finds all available timeblocks for a given date
+
+    const currentDate = DateTime.now().setZone('America/Toronto');
+    let nextHour = '00';
+
+    if (currentDate.toISODate() === selectedDate) {
+        nextHour = String(currentDate.hour + 1);
+
+        if (nextHour.length === 1) {
+            nextHour = `0${nextHour}`;
+        }
+    }
+
+    const startDate = DateTime.fromISO(String(`${selectedDate}T${nextHour}:00`), { zone: 'America/Toronto' });
+    const endDate = DateTime.fromISO(String(`${selectedDate}T23:00`), { zone: 'America/Toronto' });
+
+    const bookedTimeblocks = await Timeblock.find({
+        sectionId: sectionInformation._id,
+        startsAt: { $gte: startDate.toJSDate() },
+        endsAt: { $lte: endDate.toJSDate() },
+    });
+
+    const timeBlocks = [];
+    let currHourStart = startDate;
+    let currHourEnd = currHourStart.plus({ hours: 1 });
+
+    // Iterate through the hours in the given time range
+    while (currHourStart < endDate) {
+        // Finds schedule start and end for the day that the current hour falls on
+        const currWeekDay = currHourStart.weekday;
+        let scheduleDayStart = 0;
+        let scheduleDayEnd = 0;
+        for (const day of roomInformation.schedule) {
+            if (day.dayOfWeek + 1 === currWeekDay) {
+                scheduleDayStart = day.start;
+                scheduleDayEnd = day.end;
+            }
```

Comment:
You can replace this block with something like this:
```ts
const { start: scheduleDayStart, end: scheduleDayEnd }  = roomInformation.schedule.find(day => day.dayOfWeek + 1 === currWeekDay) ?? { start: 0, end: 0 }
```

This uses destructuring to set the variables `scheduleDayStart` and `scheduleDayEnd`, and sets both to 0 if the day was not found in the schedule.